arch/sim/src/sim/up_blocktask.c

```diff
@@ -118,7 +118,11 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
        * value, then this is really the previously running task restarting!
        */
 
+#ifdef CONFIG_SIM_PREEMPTIBLE
```

Comment:
It's better to combine this patch with:
https://github.com/apache/incubator-nuttx/pull/1009/commits/0e2bee8c4731ffa616a48e270be346947a4b489b